packages/@uppy/dashboard/src/index.js

```diff
@@ -658,6 +658,18 @@ module.exports = class Dashboard extends Plugin {
     }
   }
 
+  handleRestored = () => {
+    // Hide and show files to trigger re-render, so that
+    // VirtualList mounts FileItems again and they emit `thumbnail:request`
+    // Otherwise thumbnails are broken or missing after Golden Retriever restores files
+    if (this.opts.showSelectedFiles) {
+      this.setOptions({ showSelectedFiles: false })
+      setTimeout(() => {
+        this.setOptions({ showSelectedFiles: true })
+      }, 4)
+    }
+  }
+
```

Comment:
@goto-bus-stop is there a more sane way to trigger re-render of VirtualList to get thumnails re-created? 🤔

Reply:
We could add a `componentDidUpdate()` handler to the `FileItem` component that requests a new thumbnail when the value of `isGhost` changed from `true` to `false`

Reply:
Good idea, but files lose previews not because of ghosts, but because of object.url becoming invalid/inaccessible. Hence I need to re-generate previews for all restored files. Maybe `isRestored` is the way to go.